### ISSUE TITLE
Accessibility Feedback - Staging Review - Design System Team - Modal Web Component

### ISSUE THREAD

**Issue by briandeconinck** [department-of-veterans-affairs/va.gov-team#38379]
## VFS acceptance criteria
- [ ] Let Platform know when the **Must** feedback has been incorporated 
- [ ] Leave any comments for feedback you decide _not_ to take
- [ ] VFS team closes the ticket

## Thoughts/questions
-

## Feedback
- **Must** = the feedback must be applied
- **Should** = the feedback is best practice
- **Consider** = suggestions/enhancements

**Must:**
All content must be viewable at high zoom. Right now the modal is fine up to 200%, but beyond that you start getting some content cut off across the top that can't be scrolled to. It should work up to roughly 400% zoom (see [WCAG 1.4.10](https://www.w3.org/WAI/WCAG21/Understanding/reflow.html) for a discussion on where 400% comes from, if you're curious!) Note that the [existing React component](https://design.va.gov/storybook/?path=/docs/components-modal--default) is fine at 400%.

Screenshot of web component modal at 250% zoom, with content cut off across the top: 
![modal-zoom-issue](https://user-images.githubusercontent.com/6844775/157736693-0e89bf7d-e2bb-4958-9256-bddfb84c1105.png)

**Should:**

**Consider:**
It sounds like Courtney is already discussing this with you, but documenting here anyway --- Consider removing the `hide-close-button` property from this component. As a general rule, modals should have at least two ways to close them: using the `Esc` key, and a visible keyboard-focusable control. Toggling `hide-close-button` to true doesn't necessarily violate this, so long as either the primary action or secondary action button is used to close the modal. But that introduces some inconsistency of user interaction depending on how VFS teams choose to use the component. I think the better option would be to just remove `hide-close-button` from the component and ensure all modals have that close button.

## Platform directions
- Update "Issue Title"
- Link to collab cycle touchpoint meeting epic
- Add to the Feedback pipeline on the Platform Governance team board
- Add your feedback
- Add assignees based on collab cycle touchpoint
  - **Design Intent**: VFS designer, VFS PM (optional), yourself (optional)
  - **Midpoint Review**: VFS PM, yourself (optional)
  - **Staging Review QA Only**: VFS PM, yourself (optional)

**Comment by andresriveratoro** [comment]
@briandeconinck Thank you for taking the time to document this and for providing that WCAG link. I've updated the component to behave the same way as the React component at 400% zoom and removed the ability to hide the close button to provide a more predictable experience. You can see these changes here: https://github.com/department-of-veterans-affairs/component-library/pull/298#issuecomment-1064440433 Thanks again!

**Comment by briandeconinck** [comment]
@andresriveratoro Awesome, and thank you!
